colossus/src/main/scala/colossus/core/Connection.scala

```diff
@@ -179,7 +184,7 @@ private[core] class ServerConnection(id: Long, key: SelectionKey, channel: Socke
     super.close(cause)
   }
 
-  def isTimedOut(time: Long) = server.maxIdleTime != Duration.Inf && time - lastTimeDataReceived > server.maxIdleTime.toMillis
+  def isTimedOut(time: Long) = server.maxIdleTime != Duration.Inf && time - math.max(lastTimeDataReceived, lastTimeDataWritten) > server.maxIdleTime.toMillis
```

Comment:
The 'isTimedOut' function is identical in both ServerConnection and ClienConnection.  The only difference being of where the 'maxIdleTime' comes from.  Can we make a function on Connection:

isTimedOut(maxIdleTime : Duration, time : Long) =  maxIdleTime != Duration.Inf && time - math.max(lastTimeDataReceived, lastTimeDataWritten) > maxIdleTime.toMillis

This would just move this code into a common location